unit_tests/test_class.py

```diff
@@ -0,0 +1,166 @@
+    assert gr.__dict__['end'] == 'p12.2'
+
+    with pytest.raises(AssertionError):
+        TranscriptComponent(transcript = 'nm152263.3', exon_start = '1', exon_start_offset = '-9',
+                            exon_end = '8', exon_end_offset = '7', gene = 4,
+                            component_genomic_region = 5)
+
+def test_critical_domain():
+    """Test CriticalDomain object initializes correctly"""
+    gen1 = GeneDescriptor(value_id='hgnc:1', label='G1')
+    domain = CriticalDomain(status='preserved', name='tyrosine kinase catalytic domain', id='interpro:IPR020635',
+                            gene=gen1)
+
+    assert domain.__dict__['status'] == 'preserved'
+    assert domain.__dict__['name'] == 'tyrosine kinase catalytic domain'
+    assert domain.__dict__['id'] == 'interpro:IPR020635'
+    gen = domain.__dict__['gene']
+    assert gen.__dict__['value_id'] == 'hgnc:1'
+    assert gen.__dict__['label'] == 'G1'
+
+    with pytest.raises(AssertionError):
+        CriticalDomain(status='gained', name= 'tyrosine kinase catalytic domain', id='interproIPR020635',
+                            gene= 7)
+
+def test_fusion():
+    """Test Fusion Object initializes correctly"""
+    gen1 = GeneDescriptor(value_id='hgnc:12012', label='TPM3')
+    gen2 = GeneDescriptor(value_id='hgnc:8031', label='NTRK1')
+    reg1 = RegulatoryElement(type='promoter', value_id='hgnc:100', label='ASIC1')
+    reg2 = RegulatoryElement(type='enhancer', value_id='hgnc:200', label='G1')
+
+    domain = CriticalDomain(status='preserved', name='tyrosine kinase catalytic domain', id='interpro:IPR020635',
+                            gene=gen2)
+
+    ur1 = SequenceLocation(type='SequenceLocation', seq_id='ncbi:NC_000001.11', start=1000, end=5000)
+    c1 = UnknownGene(region=ur1)
+    ur2 = ChromosomeLocation(type='ChromosomeLocation', species='taxonomy:9606', chr='12', start='p12.1', end='p12.16')
+    c2 = UnknownGene(region=ur2)
+
+    ex1 = GenomicRegion(value=SequenceLocation(type='SequenceLocation', seq_id='ncbi:NC_000001.11', start=154170399, end=1556346346))
+    ex2 = GenomicRegion(value=SequenceLocation(type='SequenceLocation', seq_id='ncbi:NC_000001.13', start=154170401, end=1556346348))
+
+    tr1 = TranscriptComponent(transcript='nm:152263.3', exon_start=1, exon_start_offset=-9,
+                              exon_end=8, exon_end_offset=7, gene=gen1, component_genomic_region=ex1)
+    tr2 = TranscriptComponent(transcript='nm:002529.3', exon_start=10, exon_start_offset=-8,
+                              exon_end=17, exon_end_offset=8, gene=gen2, component_genomic_region=ex2)
+    linker_seq = Linker(linker_sequence='ATTATTA')
+
+    cause_eve = Event(event_type='rearrangement')
+
+    try:
+        Fusion(r_frame_preserved = True, regulatory_elements = [reg1, reg2], protein_domains = [domain],
+                                  transcript_components = [c1, c2, tr1, linker_seq, tr2], causative_event = cause_eve)
+    except AssertionError:
+        pass
```

Comment:
If this is valid, why are we using try/except?

Reply:
I used try/except to make sure it's valid but I guess it could be removed?

Reply:
Ok. If it's expected to be valid we should remove the try/except. If for some reason it threw an error, we would want to see this when running tests.